src/Func.h

```diff
@@ -1147,6 +1147,15 @@ class Func {
      * runtime error will occur when you try to run your pipeline. */
     EXPORT Func &bound(Var var, Expr min, Expr extent);
 
+    /** Statically declare the range over which the function will be
+     * evaluated in the general case. This provides a basis for the auto
+     * scheduler to make trade-offs and scheduling decisions. The estimates will
+     * be used in conjunction with bounds if they are specified. The auto
+     * generated schedules might break when the sizes of the dimensions are very
+     * different from the estimates specified. These estimates are used only
+     * by the auto scheduler.*/
+    EXPORT Func &estimate(Var var, Expr min, Expr extent);
```

Comment:
No, it doesn't. It only uses the estimates. I added a validator phase to ensure we don't have partial bounds specified since it might mess up the auto-scheduler.